pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -161,6 +162,15 @@ func (c *Cluster) Start() error {
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
 
+			// During an upgrade, make sure the server_addr setting is cleared or else the prometheus
+			// service may not be able to bind to the pod ip anymore, as was the setting in 4.7 and prior.
+			// This is only needed for 4.8 since upgrades to later clusters will always use the default.
+			logger.Info("ensuring the prometheus server_port is unset to binding to the localhost by default")
+			_, err = client.SetConfig(c.context, c.clusterInfo, fmt.Sprintf("mgr.%s", daemonID), fmt.Sprintf("mgr/prometheus/%s/server_addr", daemonID), "", false)
+			if err != nil {
+				logger.Warningf("failed to unset the prometheus server_port. %v", err)
```

Comment:
Should this return an error since it's necessary to upgrade?

Reply:
This command is only needed once during the initial upgrade. After that, it will still run on every reconcile and be a no-op. If it fails unexpectedly, we don't really want to fail the whole reconcile since it likely already succeeded on a previous reconcile.